src/Microsoft.Graph.Core/Requests/Content/BatchRequestContent.cs

```diff
@@ -0,0 +1,198 @@
+﻿// ------------------------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All Rights Reserved.  Licensed under the MIT License.  See License in the project root for license information.
+// ------------------------------------------------------------------------------
+
+namespace Microsoft.Graph
+{
+    using Newtonsoft.Json;
+    using Newtonsoft.Json.Linq;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Net;
+    using System.Net.Http;
+    using System.Net.Http.Headers;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// A <see cref="HttpContent"/> implementation to handle json batch requests.
+    /// </summary>
+    public class BatchRequestContent: HttpContent
+    {
+        private const int MAX_NUMBER_OF_REQUESTS = 20;
+
+        /// <summary>
+        /// A BatchRequestSteps property.
+        /// </summary>
+        public IDictionary<string, BatchRequestStep> BatchRequestSteps { get; private set; }
```

Comment:
I think you want IReadOnlyDictionary there.  Allowing folks to bypass the the AddBatchRequestStep could be problematic.